### PR TITLE
feat(office): Preview PDF rep if using internet explorer

### PR DESCRIPTION
**Pull request by ConradJChan**
TODO
- [x] unit tests
- [x] cross browser testing

Screen Shot 2021-08-04 at 4 35 11 PM

### CODE REVIEW

File: src/lib/Preview.js [hunk 1]
```diff
@@ -1009,6 +1010,21 @@ class Preview extends EventEmitter {
...
+    }
+
+    /**
+     * Adds Office viewer to the disabled viewer list if browser is IE
+     * @returns {void}
+     */
+    forceDisableOfficeOnline() {
+        const isIE = this.browserInfo.name === BROWSERS.INTERNET_EXPLORER;
+
+        if (isIE) {
```
Comment (jstoffan) [inline src/lib/Preview.js:1025]: Can we move this logic to OfficeLoader? We have other conditions that do the same thing there.

File: src/lib/Preview.js [hunk 2]
```diff
@@ -1009,6 +1010,21 @@ class Preview extends EventEmitter {
...
+        // Disable office online if browser is IE
+        this.forceDisableOfficeOnline();
+    }
+
+    /**
+     * Adds Office viewer to the disabled viewer list if browser is IE
+     * @returns {void}
+     */
+    forceDisableOfficeOnline() {
+        const isIE = this.browserInfo.name === BROWSERS.INTERNET_EXPLORER;
```
Comment (jstoffan) [inline src/lib/Preview.js:1023]: I think we have a helper method for this somewhere.

File: src/lib/viewers/doc/DocLoader.js [hunk 3]
```diff
@@ -18,6 +19,12 @@ const VIEWERS = [
         REP: 'pdf',
         EXT: ['gslide', 'gslides', 'odp', 'ppt', 'pptx', 'key'],
     },
+    {
+        NAME: 'OfficeExcel',
+        CONSTRUCTOR: OfficeExcelViewer,
```
Comment (jstoffan) [inline src/lib/viewers/doc/DocLoader.js:24]: Is this needed? We should already support this fallback path today?

File: src/lib/viewers/doc/OfficeExcelViewer.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import DocumentViewer from './DocumentViewer';
+
+class OfficeExcelViewer extends DocumentViewer {
+    /**
+     * @inheritdoc
+     */
+    load() {
+        super.load();
+
+        this.options.ui.showNotification(__('error_internet_explorer_office_online'), null, true);
```
Comment (jstoffan) [inline src/lib/viewers/doc/OfficeExcelViewer.js:10]: Would we only want to show this to users who are using IE11?

File: src/lib/viewers/doc/DocBaseViewer.js [hunk 5]
```diff
@@ -377,6 +379,14 @@ class DocBaseViewer extends BaseViewer {
         this.showPreload();
 
         const template = this.options.representation.content.url_template;
+        const { extension } = this.options.file;
```
Comment (jstoffan) [inline src/lib/viewers/doc/DocBaseViewer.js:382]: Is `this.options.file` always defined as an object? Should we move this logic to DocumentViewer?

File: src/lib/viewers/doc/DocBaseViewer.js [hunk 6]
```diff
@@ -377,6 +379,14 @@ class DocBaseViewer extends BaseViewer {
...
+        const isExcelExtension = EXCEL_EXTENSIONS.indexOf(extension) !== -1;
+        const isIE = Browser.getName() === BROWSERS.INTERNET_EXPLORER;
```
Comment (jstoffan) [inline src/lib/viewers/doc/DocBaseViewer.js:384]: Should we add an `isIE()` helper to our Browser util?

File: src/lib/viewers/doc/DocBaseViewer.js [hunk 7]
```diff
@@ -377,6 +379,14 @@ class DocBaseViewer extends BaseViewer {
         this.showPreload();
 
         const template = this.options.representation.content.url_template;
+        const { extension } = this.options.file;
+        const isExcelExtension = EXCEL_EXTENSIONS.indexOf(extension) !== -1;
```
Comment (jstoffan) [inline src/lib/viewers/doc/DocBaseViewer.js:383]: Can we use `.includes()` here?

File: src/lib/viewers/doc/DocBaseViewer.js (hunk 5 above)
Comment (ConradJChan) [reply src/lib/viewers/doc/DocBaseViewer.js:382]: Seems like it is always passed into the viewer options: https://github.com/box/box-content-preview/blob/19722e68d00a62b588ef06f7fe824a55afa0e107/src/lib/Preview.js#L1211

**Review by jstoffan — APPROVED** [review]

**Pull request merged**